src/main/scala/it/pagopa/interop/partyprocess/service/impl/PDFCreatorImpl.scala

```diff
@@ -44,20 +46,42 @@ object PDFCreatorImpl extends PDFCreator with PDFManager {
     }
   }
 
-  private def setupData(manager: User, users: Seq[User], institution: Institution): Try[Map[String, String]] = {
+  private def setupData(
+    manager: User,
+    users: Seq[User],
+    institution: Institution,
+    onboardingRequest: OnboardingSignedRequest
+  ): Try[Map[String, String]] = {
     for {
       managerEmail <- manager.email.toTry("Manager email not found")
     } yield Map(
-      "institutionName"    -> institution.description,
-      "institutionTaxCode" -> institution.taxCode,
-      "originId"           -> institution.originId,
-      "institutionMail"    -> institution.digitalAddress,
-      "managerName"        -> manager.name,
-      "managerSurname"     -> manager.surname,
-      "managerTaxCode"     -> manager.taxCode,
-      "managerEmail"       -> managerEmail,
-      "manager"            -> userToText(manager),
-      "delegates"          -> delegatesToText(users)
+      "institutionName"      -> onboardingRequest.institutionUpdate
+        .flatMap(_.description)
+        .getOrElse(institution.description),
+      "institutionTaxCode"   -> onboardingRequest.institutionUpdate.flatMap(_.taxCode).getOrElse(institution.taxCode),
+      "originId"             -> institution.originId,
+      "institutionMail"      -> institution.digitalAddress,
+      "managerName"          -> manager.name,
+      "managerSurname"       -> manager.surname,
+      "managerTaxCode"       -> manager.taxCode,
+      "managerEmail"         -> managerEmail,
+      "manager"              -> userToText(manager),
+      "delegates"            -> delegatesToText(users),
+      "institutionType"      -> transcodeInstitutionType(
+        onboardingRequest.institutionUpdate
+          .flatMap(_.institutionType)
+          .orElse(institution.institutionType)
+          .getOrElse("")
+      ),
```

Comment:
`transcodeInstitutionType` does not manage empty string values so an error will occur here.

Reply:
institutionType is supposed to be defined as mandatory between request and institution ;)

Reply:
Ok, but here we are dealing with two Option values, so there is nothing mandatory. :) 
Maybe you're talking about what actually should happen, but the code knows nothing about that.

This implementation 
``` 
private def transcodeInstitutionType(institutionType: String): String = {
    institutionType.toUpperCase match {
      case "PA"  => "Pubblica Amministrazione"
      case "GSP" => "Gestore di servizi pubblici"
      case "SCP" => "Società a controllo pubblico"
      case "PT"  => "Partner tecnologico"
    }
  }
```
will fail if accidentally an empty string is passed.
To prevent this unsafe behavior, you can simply work on optional side like:
```
      "institutionType"      -> onboardingRequest.institutionUpdate
        .flatMap(_.institutionType)
        .orElse(institution.institutionType)
        .map(transcodeInstitutionType)
        .getOrElse(""),
```

Reply:
done :)